### PR TITLE
boards: snps: Add nSIM RISC-V platform support

### PR DESCRIPTION
**Pull request by kokas-a**
Add basic support for Synopsys nSIM RISC-V platforms. RMX500 board was added as an instance of such platform.

### CODE REVIEW

**Comment by kokas-a** [comment]
@evgeniy-paltsev, @abrodkin, @ruuddw, Here is one more attempt to add nsim_riscv platform. Add your feedback please.

File: boards/snps/nsim_riscv/nsim_riscv_rmx500.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+identifier: nsim_riscv/rmx500
+name: Synopsys rmx500
+simulation: nsim
+simulation_exec: nsimdrv
+type: sim
+arch: riscv
+toolchain:
+  - zephyr
+  - cross-compile
+  - arcmwdt
```
Comment (evgeniy-paltsev) [inline boards/snps/nsim_riscv/nsim_riscv_rmx500.yaml:10]: I guess we don't support mwdt right now?

File: boards/snps/nsim_riscv/nsim_riscv_rmx500_defconfig [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_XIP=n
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_BUILD_OUTPUT_HEX=y
+CONFIG_MP_MAX_NUM_CPUS=1
+#CONFIG_FLASH_SIZE=0
+#CONFIG_FLASH_BASE_ADDRESS=0x0
```
Comment (evgeniy-paltsev) [inline boards/snps/nsim_riscv/nsim_riscv_rmx500_defconfig:11]: Do we need this commented stuff?

File: boards/snps/nsim_riscv/nsim_riscv_rmx500_defconfig [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2024 Synopsys, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_XIP=n
+CONFIG_CONSOLE=y
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+CONFIG_BUILD_OUTPUT_HEX=y
+CONFIG_MP_MAX_NUM_CPUS=1
```
Comment (evgeniy-paltsev) [inline boards/snps/nsim_riscv/nsim_riscv_rmx500_defconfig:9]: Isn't it 1 by default when SMP=n? Probably we don't need it.

File: boards/snps/nsim_riscv/nsim_riscv_rmx500_defconfig (hunk 2 above)
Comment (kokas-a) [reply boards/snps/nsim_riscv/nsim_riscv_rmx500_defconfig]: No, we don't. Fixed

File: boards/snps/nsim_riscv/nsim_riscv_rmx500_defconfig (hunk 3 above)
Comment (kokas-a) [reply boards/snps/nsim_riscv/nsim_riscv_rmx500_defconfig]: Done

File: boards/snps/nsim_riscv/nsim_riscv_rmx500.yaml (hunk 1 above)
Comment (kokas-a) [reply boards/snps/nsim_riscv/nsim_riscv_rmx500.yaml]: Fixed

File: soc/snps/nsim_riscv/rmx500/Kconfig.soc [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2024 Synopsys, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_NSIM_RV_RMX500
+	bool
+	select SOC_FAMILY_NSIM_RV
+
+config SOC_SERIES
+	default "rmx500" if SOC_SERIES_NSIM_RV_RMX500
```
Comment (nordicjm) [inline soc/snps/nsim_riscv/rmx500/Kconfig.soc:9]: Needs to match e.g. `default "nsim_rv_rmx500" if SOC_SERIES_NSIM_RV_RMX500`

File: soc/snps/nsim_riscv/rmx500/Kconfig.soc [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+config SOC_NSIM_RV_RMX500
+	bool
+	select SOC_SERIES_NSIM_RV_RMX500
+
+config SOC
+	default "rmx500" if SOC_NSIM_RV_RMX500
```
Comment (nordicjm) [inline soc/snps/nsim_riscv/rmx500/Kconfig.soc:16]: as above

File: soc/snps/nsim_riscv/Kconfig.soc [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) 2024 Synopsys, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_NSIM_RV
+	bool
+
+config SOC_FAMILY
+	default "nsim_riscv" if SOC_FAMILY_NSIM_RV
```
Comment (nordicjm) [inline soc/snps/nsim_riscv/Kconfig.soc:8]: as above

File: soc/snps/nsim_riscv/rmx500/Kconfig.soc (hunk 4 above)
Comment (nordicjm) [reply soc/snps/nsim_riscv/rmx500/Kconfig.soc:9]: and needs to match the contents of soc.yml

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by kokas-a** [comment]
Dublicate with https://github.com/zephyrproject-rtos/zephyr/pull/74171

**Pull request closed without merging**